### PR TITLE
moving around external annotators: putting everything in a folder cal…

### LINKED ISSUE #468: move external resources as submodules  (instead of modules)
Currently a few resources are included as single module `external` in addition to `stanford_3.3.1`: 
```
├── external
│   ├── pom.xml
├── stanford_3.3.1
│   ├── pom.xml
```
These two are not merged as one single module, since 

I wanted to add a few more annotators, but it's gonna clutter our project structure if we have too many of these non-Cogcomp modules. Here is my suggested modifications to the current structure. Have a **folder** `external` (rather than a module) which contains submodules for each annotator: 
```
├── external
├──── path-lstm
│   ├──── pom.xml
├──── stanford_3.7.0
│   ├──── pom.xml
├──── stanford_3.3.1
│   ├──── pom.xml
├──── rst-parser
│   ├──── pom.xml
```

what do you think @mssammon ?

### PR DESCRIPTION
**Pull request by danyaljj**
Changes to external annotators: Following the conversation in #468, 
 - Everything is moved inside a folder "external", so that they don't clutter our standard libraries. All the external annotators will be sub-modules of this folder. 
 - A module named "common" created to prevent repetition. Also, they no-longer depend on the pipeline or core-utils. 
 - Previously-named module "external" is separated into two sub-modules "stanford_380" and "path-lstm". 

Closes #468.

### CODE REVIEW

File: external/external-commons/pom.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        <artifactId>illinois-cogcomp-nlp</artifactId>
+        <groupId>edu.illinois.cs.cogcomp</groupId>
+        <version>3.1.25</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>external-commons</artifactId>
+    <name>Eexternal Annotators Commons</name>
```
Comment (mssammon) [inline external/external-commons/pom.xml]: typo 'Eexternal'

File: external/external-commons/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/ExternalAnnotatorsServer.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static spark.Spark.*;
+
+public class ExternalAnnotatorsServer {
```
Comment (mssammon) [inline external/external-commons/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/ExternalAnnotatorsServer.java]: Here and elsewhere, please add a Javadoc with basic description

File: external/external-commons/pom.xml (hunk 1 above)
Comment (danyaljj) [inline external/external-commons/pom.xml]: fixed.

File: external/external-commons/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/ExternalAnnotatorsServer.java (hunk 2 above)
Comment (danyaljj) [inline external/external-commons/src/main/java/edu/illinois/cs/cogcomp/pipeline/server/ExternalAnnotatorsServer.java]: done.

**Comment by danyaljj** [comment]
Addressed the comments. Any other suggestions?

**Comment by mssammon** [comment]
Looks good to me.

**Pull request merged**